src/main/java/com/bobocode/nasapicturesstealer/service/NasaPictureReceiver.java

```diff
@@ -0,0 +1,67 @@
+package com.bobocode.nasapicturesstealer.service;
+
+import com.bobocode.nasapicturesstealer.configuration.NasaUrlProperty;
+import com.bobocode.nasapicturesstealer.service.model.NasaCamera;
+import com.bobocode.nasapicturesstealer.service.model.NasaPicture;
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.AllArgsConstructor;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+
+@AllArgsConstructor
+@Service
+public class NasaPictureReceiver {
+
+    private final RestTemplate restTemplate;
+    private final NasaUrlProperty nasaUrlProperty;
+
+    @Cacheable("receiveAllPictures")
```

Comment:
When you will evict cache? Do third party provider can't add photo for specific sol?

Reply:
Added the `@CacheEvict` and `@Scheduled`. I think the third party adds photo rare and we can handle it.